packages/e2e-tests/specs/plugins/inner-blocks-renderappender.test.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	activatePlugin,
+	createNewPost,
+	deactivatePlugin,
+	getAllBlockInserterItemTitles,
+	getEditedPostContent,
+	insertBlock,
+	openAllBlockInserterCategories,
+} from '@wordpress/e2e-test-utils';
+
+const QUOTE_INSERT_BUTTON_SELECTOR = '//button[contains(concat(" ", @class, " "), " block-editor-block-types-list__item ")][./span[contains(text(),"Quote")]]';
```

Comment:
If it was really the case that the inner span selection ended up being an issue resolved in #18771, then I think we should be a bit more wary about it. I think there might be some options with XPath to traverse back up to the parent of a matched span, if we need to match it? https://devhints.io/xpath#more-examples

Generally, I kinda wish these elements were just easier to target. I'd almost be okay with adapting the output in the source code for the sake of the tests, if it makes them more reliable.

Reply:
I updated the selection mechanism to use logic similar to what we used on https://github.com/WordPress/gutenberg/pull/18771.